core/sql/optimizer/OptimizerSimulator.cpp

```diff
@@ -278,9 +277,10 @@ OptimizerSimulator::OptimizerSimulator(CollHeap *heap)
     sysCallLogFilePath_[sc]=NULL;
     writeSysCallStream_[sc]=NULL;
   }
+  // Init log4cxx logging
+  CommonLogger::instance().initLog4cxx("log4cxx.trafodion.osim.config");
```

Comment:
Hi Howard, thanks for the explanation. Can I ask two more questions? First, if we do a CONTROL OSIM command, then we will go through this code. That probably means that all subsequent log messages (after the OSIM command completes) for this process will appear on the screen and won't be logged to files anymore? Also, if people do the OSIM command from trafci or from a JDBC client, they would not see the console log?

Assuming I understand this correctly, here is one possible solution: Don't use a logger. Instead, use printf (or similar) to log to a file, along with the other information. Then issue a warning in the OSIM command that looks like this: **** WARNING 1234: Log output has been written to /somedir/osim_log_20150707_1234.log. For debugging OSIM code, you can just use the regular logger and examine the log files. Also, you can use an #ifdef OSIM_DEBUG or similar if you want to print debugging info to the screen.

Again, I hope I understood the problem well enough that what I say above makes sense.